app/lib/account/models.dart

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:gcloud/db.dart' as db;
+
+/// User data model.
+@db.Kind(name: 'User', idType: db.IdType.String)
+class User extends db.ExpandoModel {
+  String get userId => id as String;
+
+  @db.StringProperty()
+  String authProvider;
+
+  @db.StringProperty()
+  String authUserId;
+
+  @db.StringProperty()
+  String email;
+
+  @db.StringListProperty()
+  List<String> historicalEmails = <String>[];
+
+  @db.BoolProperty()
+  bool hideEmail = false;
+
+  @db.DateTimeProperty()
+  DateTime created;
+
+  @db.DateTimeProperty()
+  DateTime updated;
+
+  @db.DateTimeProperty()
+  DateTime lastAccess;
+
+  void updateEmail(String email) {
+    if (this.email != email) {
+      this.email = email;
+    }
+    if (!historicalEmails.contains(email)) {
+      historicalEmails.add(email);
+    }
+  }
+}
+
+/// Derived data for [User] for fast lookup.
+@db.Kind(name: 'UserInfo', idType: db.IdType.String)
+class UserInfo extends db.ExpandoModel {
+  String get userId => id as String;
```

Comment:
I'm tempted to suggest a static method:
```dart
static db.Key key(String userId) => db.emptyKey.append(User, id: userId);
```

Then in the code we can just write:
```dart
User user = (await db.lookup<User>([User.key(myUserId)])).single;
```

Reply:
Feel free to ignore this idea... it's purely syntactic nonsense, hehe :)

Reply:
`User.key` is already taken by the `db.ExpandoModel`. `User.keyOf` could work like this:
````
  static db.Key keyOf(String userId) =>
      db.dbService.emptyKey.append(User, id: userId);

````
However, this will access `db.dbService`, without any contextual separation. I don't think it is a concern right now, just wanted to check with you.